src/Circle.cpp

```diff
@@ -12,6 +12,61 @@ Circle Circle::Unit()
 {
     return Circle(0, 0, 1);
 }
+
+std::vector<base::Vector2d> Circle::intersect(const Eigen::ParametrizedLine<double, 2>& line) const
+{
+    //Variable names correspond to names in:
+    //http://mathworld.wolfram.com/Circle-LineIntersection.html
+    
+    std::vector<base::Vector2d> result;
+    
+    Eigen::Vector2d p1 = line.pointAt(0);
+    Eigen::Vector2d p2 = line.pointAt(1);
+    
+    //move points to coordinate system of circle
+    p1 -= center;
+    p2 -= center;
+    
+    const double dx = p2.x() - p1.x();
+    const double dy = p2.y() - p1.y();
+    const double dr2 = dx * dx + dy * dy;
+    const double D = p1.x() * p2.y() - p2.x() * p1.y();
```

Comment:
`const double D = p1.x() * d.y() - d.x() * p1.y();`
because determinant is the same for `| p1-c, p2-c |`  and `|p1-c, (p2-c)-(p1-c)| = |p1-c, d|`